### PR TITLE
SA1107 diagnostic + tests

### LINKED ISSUE #41: SA1107: CodeMustNotContainMultipleStatementsOnOneLine
http://www.stylecop.com/docs/SA1107.html

### PR DESCRIPTION
**Pull request by pdelvo**
Let me know if the tests I wrote are enough. We unfortunatly can't test fix all providers yet. This one worked in my tests. The batch fixer didn't work if more then one violation was on a single line. Fixes #41.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs [hunk 1]
```diff
@@ -43,7 +46,27 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+            {
+                for (int i = 1; i < block.Statements.Count; i++)
+                {
+                    StatementSyntax previousStatement = block.Statements[i - 1];
+                    StatementSyntax currentStatement = block.Statements[i];
+
+                    if (previousStatement.GetLocation().GetLineSpan().EndLinePosition.Line
+                        == currentStatement.GetLocation().GetLineSpan().StartLinePosition.Line)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptor, currentStatement.GetLocation()));
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs]: `currentStatement.GetLocation()`
location cached internally ?

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs [hunk 2]
```diff
@@ -43,7 +46,27 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleBlock, SyntaxKind.Block);
+        }
+
+        private void HandleBlock(SyntaxNodeAnalysisContext context)
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs]: static

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs (hunk 1 above)
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs]: It isnt. But its a relativly cheap object to create. But I fixed it anyway

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs (hunk 2 above)
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1107CodeMustNotContainMultipleStatementsOnOneLine.cs]: fixed

**Comment by pdelvo** [comment]
I fixed a bug

**Pull request merged**